front/app/modules/commercial/granular_permissions/admin/components/UserFieldSelection/FieldSelectionModal.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+// components
+import { Box, Title } from '@citizenlab/cl2-component-library';
+import Modal from 'components/UI/Modal';
+
+// services
+import { IUserCustomFieldData } from 'services/userCustomFields';
+
+// intl
+import { SelectionScreen } from './screens/SelectionScreen';
+import { AddFieldScreen } from './screens/AddFieldScreen';
+import { IPermissionsCustomFieldData } from 'api/permissions_custom_fields/types';
+import { FormattedMessage } from 'utils/cl-intl';
+import messages from '../../containers/Granular/messages';
+
+type FieldSelectionModalProps = {
+  showSelectionModal: boolean;
+  setShowSelectionModal: (show: boolean) => void;
+  selectedFields: Array<IPermissionsCustomFieldData> | undefined;
+  handleAddField: (field: IUserCustomFieldData) => void;
+  registrationFieldList: Array<IUserCustomFieldData> | null | undefined;
+  locale: string;
+  isLoading: boolean;
+};
+
+export const FieldSelectionModal = ({
+  showSelectionModal,
+  setShowSelectionModal,
+  selectedFields,
+  handleAddField,
+  registrationFieldList,
+  locale,
+  isLoading,
+}: FieldSelectionModalProps) => {
+  const [showAddFieldPage, setShowAddFieldPage] = React.useState(false);
+  return (
+    <Modal
+      opened={showSelectionModal}
+      close={() => {
+        setShowSelectionModal(false);
+      }}
+      header={
+        <Title variant="h3" color="primary">
+          <FormattedMessage
+            {...(showAddFieldPage
+              ? messages.createAQuestion
+              : messages.addQuestion)}
+          />
+        </Title>
+      }
+      width={'550px'}
```

Comment:
Also, this still looks a bit off:

![image](https://user-images.githubusercontent.com/13822005/231130575-3fd418b2-2bc3-4483-8c6f-5331d764fd8a.png)

I think it should be fixed by adding `niceHeader={true}` to the `Modal` component, but I also still made some small changes to this `Modal` component on my branch, so it might not work optimally yet on this version

Reply:
The niceHeader={true} works! :) Didn't realize we had that attribute actually - thanks!!